pkg/querier/queryrange/downstreamer_test.go

```diff
@@ -598,3 +600,70 @@ func TestDownstreamAccumulatorMultiMerge(t *testing.T) {
 		})
 	}
 }
+
+func newRandomSketch() sketch.QuantileSketch {
+	r := rand.New(rand.NewSource(42))
+	s := sketch.NewDDSketch()
+	for i := 0; i < 1000; i++ {
+		_ = s.Add(r.Float64())
+	}
+	return s
+}
+
+func BenchmarkAccumulateMerge(b *testing.B) {
+	b.ReportAllocs()
+	results := make([]logql.ProbabilisticQuantileVector, 100)
+	for i := range results {
+		results[i] = make(logql.ProbabilisticQuantileVector, 100)
+		for j := range results[i] {
+			results[i][j] = logql.ProbabilisticQuantileSample{
+				T:      int64(i),
+				F:      newRandomSketch(),
+				Metric: []labels.Label{{Name: "foo", Value: fmt.Sprintf("bar-%d", j)}},
+			}
+		}
+	}
+	res := make([]logqlmodel.Result, 0, 100)
+	for _, r := range results {
+		res = append(res, logqlmodel.Result{
+			Data: logql.ProbabilisticQuantileMatrix([]logql.ProbabilisticQuantileVector{r}),
+		})
+	}
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		acc := newQuantileSketchAccumulator()
+		for _, r := range res {
+			require.NoError(b, acc.Accumulate(r, 0))
+		}
+	}
+}
```

Comment:
Nice. I was about to push my change and had a conflict. Your code was almost exactly as mine :tada: